salt/modules/pkg_resource.py

```diff
@@ -27,11 +27,22 @@ def _repack_pkgs(pkgs, normalize=True):
         _normalize_name = __salt__["pkg.normalize_name"]
     else:
         _normalize_name = lambda pkgname: pkgname
-    return {
+
+    repacked_pkgs = {
         _normalize_name(str(x)): str(y) if y is not None else y
         for x, y in salt.utils.data.repack_dictlist(pkgs).items()
     }
 
+    # Check if there were collisions in names
+    if len(pkgs) != len(repacked_pkgs):
+        raise CommandExecutionError(
```

Comment:
This would be a `SaltInvocationError` as it's a problem with passing invalid input to the command. The `CommandExecutionError` is when there is an error when salt tries to make a change to the system.

Reply:
Solved. Thanks for the hint!